src/problem.ts

```diff
@@ -1,62 +1,78 @@
-import { httpObject, ProblemInterface } from "types";
+import { DEFAULT_KEYS } from "./constants";
+import {
+  httpObject,
+  ProblemInterface,
+  ToJsonParamType,
+  UpdateProblemParamType,
+} from "./types";
 
-enum COPY_MODE {
-    SKIP_PROPS = 'skipProps',
-    LEAVE_PROPS = 'leaveProps'
+export enum COPY_MODE {
+  SKIP_PROPS = "skipProps",
+  LEAVE_PROPS = "leaveProps",
 }
 
 export class Problem extends Error implements ProblemInterface {
-    public type: string;
-    public title: string;
-    public instance?: string;
-    public detail?: string;
-    public http?: httpObject
-    [key: string]: any;
-
-
-    constructor(problem: ProblemInterface, customKeys?: string[]) {
-        super(problem.detail || problem.title);
-        this.http = problem.http
-        this.type = problem.type
-        this.title = problem.title;
-        this.detail = problem.detail;
-        this.instance = problem.instance;
-        this.stack = problem.stack;
-        customKeys?.map((customKey) => {
-            this[customKey] = problem[customKey];
-        })
-    }
+  public type: string;
+  public title: string;
+  public instance?: string;
+  public detail?: string;
+  public http?: httpObject;
+  [key: string]: any;
 
-   copy(problem: ProblemInterface, mode: COPY_MODE, props: string[]): Problem {
-        switch (mode) {
-
-            case COPY_MODE.LEAVE_PROPS:
-                return new Problem(problem, props);
-
-            case COPY_MODE.SKIP_PROPS:
-            default:
-                let keysToBeCopied: string[] = [];
-                for (let key in problem) {
-                    if (props.includes(key))
-                        continue;
-                    keysToBeCopied.push(key)
-                }
-                return new Problem(problem, keysToBeCopied)
-        }
-    };
+  constructor(problem: ProblemInterface, customKeys?: string[]) {
```

Comment:
We should change constructor to this:

```suggestion
  constructor(
    protected readonly problem: ProblemInterface,
  ) {
```

By this we can reuse `this.problem` in `copy method` and `customKeys` as I wrote in previous review, aren't necessary.

Reply:
I am thinking out loud, but maybe rather:

```ts
  public type: string;
  public title: string;
  public instance?: string;
  public detail?: string;
  public http?: httpObject;
  [key: string]: any;
```

should we have `get(key: ...)` function to retrieve `title`, `detail` etc. and custom keys? What do you think? :)